samples/sample_common/src/vaapi_allocator.cpp

```diff
@@ -332,7 +333,26 @@ mfxStatus vaapiFrameAllocator::AllocImpl(mfxFrameAllocRequest *request, mfxFrame
             {
                 format = VA_RT_FORMAT_RGBP;
             }
+#ifdef ENABLE_V4L2_SUPPORT
+            if ((m_export_mode & vaapiAllocatorParams::PRIME) && (format == VA_RT_FORMAT_YUV422))
+            {
+                VASurfaceAttribExternalBuffers buffer_desc;
+                buffer_desc.pixel_format        = va_fourcc;
+                buffer_desc.width               = request->Info.Width;
+                buffer_desc.height              = request->Info.Height;
+                buffer_desc.flags               = 0;
+
+                attrib[attrCnt].type            = VASurfaceAttribMemoryType;
+                attrib[attrCnt].flags           = VA_SURFACE_ATTRIB_SETTABLE;
+                attrib[attrCnt].value.type      = VAGenericValueTypeInteger;
+                attrib[attrCnt++].value.value.i = VA_SURFACE_ATTRIB_MEM_TYPE_VA;
 
+                attrib[attrCnt].type            = (VASurfaceAttribType)VASurfaceAttribExternalBufferDescriptor;
+                attrib[attrCnt].flags           = VA_SURFACE_ATTRIB_SETTABLE;
+                attrib[attrCnt].value.type      = VAGenericValueTypePointer;
+                attrib[attrCnt++].value.value.p = &buffer_desc;
+            }
+#endif
```

Comment:
just a question: does it force linear memory type?

Reply:
Yep, we passed part of description with VA_SURFACE_ATTRIB_MEM_TYPE_VA  for force user description and disable tiling:

https://github.com/intel/media-driver/blob/7857d6bea214b755d610c1a478d4e1e23da31cdb/media_driver/linux/common/ddi/media_libva.cpp#L2307

https://github.com/intel/media-driver/blob/7857d6bea214b755d610c1a478d4e1e23da31cdb/media_driver/linux/common/ddi/media_libva_util.cpp#L329

Reply:
I am looking into that and ask myself: how I should get this from the VAAPI specification that specifying this attribute will force Linear type?!

IMHO: we need add this to VAAPI specification. What you are doing here is just a workaround matching (accidentally) current driver behavior. Start creating VAAPI spec. proposal!